examples/quote_ident/src/lib.rs

```diff
@@ -0,0 +1,38 @@
+// Copyright 2018 Benjamin Fry <benjaminfry@me.com>
+//
+// Licensed under the Apache License, Version 2.0, <LICENSE-APACHE or
+// http://apache.org/licenses/LICENSE-2.0> or the MIT license <LICENSE-MIT or
+// http://opensource.org/licenses/MIT>, at your option. This file may not be
+// copied, modified, or distributed except according to those terms.
+
+extern crate pg_extend;
+extern crate pg_extern_attr;
+
+use pg_extend::guard_pg;
+use pg_extend::{pg_magic, pg_sys};
+use pg_extern_attr::pg_extern;
+
+// This tells Postgres this library is a Postgres extension
+pg_magic!(version: pg_sys::PG_VERSION_NUM);
+
+/// The pg_extern attribute wraps the function in the proper functions syntax for C extensions
+#[pg_extern]
+pub fn rust_quote_ident(value: String) -> String {
```

Comment:
As to taking and returning `String`, I've been wondering if we want to better encourage zero-overhead design patterns by only accepting and returning CStr (or CString as necessary, though I think we want to be careful passing owned data from Rust to PG).

If we do that then it becomes the responsibility of the pg-extend user to do the conversations and *see* the overhead that's being incurred with them.